backend/src/ee/routes/v1/pam-account-routers/pam-account-router.ts

```diff
@@ -125,18 +129,55 @@ export const registerPamAccountRouter = async (server: FastifyZodProvider) => {
           })
       }),
       response: {
-        200: z.object({
-          sessionId: z.string(),
-          resourceType: z.nativeEnum(PamResource),
-          relayClientCertificate: z.string(),
-          relayClientPrivateKey: z.string(),
-          relayServerCertificateChain: z.string(),
-          gatewayClientCertificate: z.string(),
-          gatewayClientPrivateKey: z.string(),
-          gatewayServerCertificateChain: z.string(),
-          relayHost: z.string(),
-          metadata: z.record(z.string(), z.string().optional()).optional()
-        })
+        200: z.discriminatedUnion("resourceType", [
+          // Gateway-based resources (Postgres)
+          z.object({
+            sessionId: z.string(),
+            resourceType: z.literal(PamResource.Postgres),
+            relayClientCertificate: z.string(),
+            relayClientPrivateKey: z.string(),
+            relayServerCertificateChain: z.string(),
+            gatewayClientCertificate: z.string(),
+            gatewayClientPrivateKey: z.string(),
+            gatewayServerCertificateChain: z.string(),
+            relayHost: z.string(),
+            metadata: z.record(z.string(), z.string().optional()).optional()
```

Comment:
perhaps we should create a zod schema for these shared gateway properties and then just `.AND` them here